### PR TITLE
Update file matching logic

### LINKED ISSUE #389: `clasp push --watch` pushes even when ignored files change
> Please search [existing issues](https://github.com/google/clasp/issues) before creating a new one.
>
> Ideally, after filing the issue, try to fix the issue with a PR!

## Expected Behavior

`clasp push --watch` only pushes when a file that will be pushed changes

## Actual Behavior

`clasp push --watch` pushes when any file changes and even pushes multiple times when multiple files change (eg. when `.git/*` files change on commits, etc)

## Steps to Reproduce the Problem

  1. run `clasp push --watch`
  1. change `README.md`
  1. All files get pushed even though the `README.md` is automatically ignored

## Specifications

  - Node version (`node -v`): v9.10.1
  - Version (`clasp -v`): 1.6.3
  - OS (Mac/Linux/Windows): Mac

### PR DESCRIPTION
**Pull request by mrk3767**
Before: All files within the project were iterated over and excluded based on .claspignore
Now: Only files specified within .clasp.json:srcPatterns or the defaults will be included and then those will be iterated over and excluded based on .claspignore

Fixes #389

### CODE REVIEW

File: src/utils.ts [hunk 1]
```diff
@@ -60,6 +60,11 @@ export function getOAuthSettings(): Promise<ClaspToken> {
     });
 }
 
+// Default settings for clasp.json
+export const DEFAULT_SETTINGS = {
+  SRC_PATTERNS: ['src/**','appsscript.json'],
```
Comment (takanakahiko) [inline src/utils.ts]: Source code is not necessarily in 'src/'.  (Ex: dist / Code.js)
It is specified by `rootDir` of `.clasp.json` .

See also : https://github.com/google/clasp#rootdir-optional

**Comment by grant** [comment]
@mrk3767 Still working on this PR?

**Comment by mrk3767** [comment]
Yes, but I'm traveling out of the country for a week 😣 I wanted to get it
posted in case anyone else wanted to use the idea while I'm gone. I managed
to get it working for my flow well enough to get my project up and running.

**Comment by grant** [comment]
OK, no worries. Happy holidays.
Just FYI, I'm working on a few changes for `@2`, so you may have merge conflicts.

**Comment by grant** [comment]
Feel free to reopen a PR when in a mergeable state.
Otherwise, please comment on issues if you want others to use your ideas while you're gone. I'm trying to keep PRs fresh.
Thanks.

**Pull request closed without merging**